src/sys/windows/tcp.rs

```diff
@@ -246,7 +246,11 @@ impl TcpStream {
     }
 
     pub fn read(&self, buf: &mut [u8]) -> io::Result<usize> {
-        self.readv(&mut [buf.into()])
+        if buf.len() > 0 {
+            self.readv(&mut [buf.into()])
```

Comment:
Would it be better to IoVec::from_bytes_mut here?

Reply:
Indeed!